### ISSUE TITLE
Party chat mentions sometimes suggest users not in the party

### ISSUE THREAD

**Issue by CuriousMagpie** [HabitRPG/habitica#14627]
### General Info
  * UUID: 6b4501bb-01d4-4887-b980-258218df5344
  * User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/[112.0.0.0](http://112.0.0.0/) Safari/537.36

### Description
From user: 
> In Party chat, using @ to mention someone: - Suggests people who *aren't* in the party as available options - Does not suggest *some* people who are in the party This happens on iOS and Web, and who is not suggested seems to rotate. I usually have to copy/paste their username and hope for the best. Seems like the party chat interface should reference a SQL table that just has the party members in it, etc.

**Note**: Habitica uses MongoDB, not SQL. A solution would need to reference party members and currently, the groups collection (which includes parties and guilds) does not have a list of all members and this is by design as some guilds are quite large. Instead, the only way to get a list of party members is to query the party ID against the users collection.

**Comment by Garvitasingh10** [comment]
@CuriousMagpie , hello I am interested in addressing this problem. Can you assign it to me?"

**Comment by CuriousMagpie** [comment]
@Garvitasingh10, consider it yours! Thank you!

**Comment by RishmitaR** [comment]
@CuriousMagpie I'm a first time contributor and would like to work on this issue, if you are still working on this issue @Garvitasingh10 then no worries, but if not could I please work on this issue?

**Comment by Garvitasingh10** [comment]
Yeah ! You  can work on it ,if u have the permit from the admin , i am no
more working on it .

**Comment by RishmitaR** [comment]
@CuriousMagpie Sorry to spam, but I do think this is a issue I could feasibly fix if it still needs to be worked on. I know it doesn't have the help wanted tag anymore but I'm throwing my hat in the ring because it has yet been closed but it has been worked on before.

**Comment by Tressley** [comment]
Hi @RishmitaR -- Thank you for your interest in this issue. This is something we're addressed on other platforms and we'll apply it to web.
